### PR TITLE
Test Examples CI

### PR DESCRIPTION
**Pull request by Nikhil-Singhal-06**
### Pull request checklist

- [ ] [Issue](https://github.com/IntelLabs/scenario_execution/issues) created that explains the change and why it's needed
- [ ] Tests are part of the PR (for bug fixes / features)
- [ ] [Docs](https://intellabs.github.io/scenario_execution/) reviewed and added / updated if needed (for bug fixes / features)
- [x] Format and Lint checks (`make check`) pass locally
- [ ] PR applies Apache 2.0 License to all code files

### Pull request type

- [x] Feature

### Current behavior
No CI test for examples

### New behavior
CI test for each example

### CODE REVIEW

File: examples/example_library/README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ source install/setup.bash
 Now, run the following command to launch the scenario:
 
 ```bash
-ros2 launch scenario_execution scenario_launch.py scenario:=examples/example_library/scenarios/example_library.osc
+ros2 launch scenario_execution_ros scenario_launch.py scenario:=examples/example_library/scenarios/example_library.osc
```
Comment (fred-labs) [inline examples/example_library/README.md:18]: this should also work with scenario_execution.

File: examples/example_scenario/README.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ source install/setup.bash
 Now, run the following command to launch the scenario:
 
 ```bash
-ros2 launch scenario_execution scenario_launch.py scenario:=examples/example_scenario/hello_world.osc
+ros2 launch scenario_execution_ros scenario_launch.py scenario:=examples/example_scenario/hello_world.osc
```
Comment (fred-labs) [inline examples/example_scenario/README.md:18]: this should also work with scenario_execution.

File: scenario_execution/scenario_execution/scenario_execution_base.py [hunk 3]
```diff
@@ -367,7 +367,7 @@ def main():
     result = scenario_execution.parse()
     if result and not args.dry_run:
         scenario_execution.run()
-    result = scenario_execution.process_results()
+    result = scenario_execution.process_results(result)
```
Comment (fred-labs) [inline scenario_execution/scenario_execution/scenario_execution_base.py:370]: process_results does not have a parameter

File: scenario_execution_control/test/scenario_execution_control_test.osc [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import osc.standard
```
Comment (fred-labs) [inline scenario_execution_control/test/scenario_execution_control_test.osc:1]: please move file to test/scenarios/

File: scenario_execution_control/test/scenario_execution_control_test.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        while not self.execute_scenario_client.wait_for_service(timeout_sec=1.0) and time.time() < end_time:
+            self.get_logger().info('Service not available, waiting again...')
+        if time.time() >= end_time:
+            rclpy.shutdown()
+            raise RuntimeError(f'Service not available after {service_timeout} seconds.')
+
+        self.req = ExecuteScenario.Request()
+
+        # Initialize subscriber for scenario status
+        self.subscription = self.create_subscription(
```
Comment (fred-labs) [inline scenario_execution_control/test/scenario_execution_control_test.py:45]: We want to test the service here. 
Please use result of service call....not the scenario status.

**Review by fred-labs — CHANGES_REQUESTED** [review]

File: scenario_execution/scenario_execution/scenario_execution_base.py (hunk 3 above)
Comment (Nikhil-Singhal-06) [reply scenario_execution/scenario_execution/scenario_execution_base.py:370]: I added the result as the parameter with the default value true.

**Review by fred-labs — CHANGES_REQUESTED** [review]

File: scenario_execution/scenario_execution/scenario_execution_base.py [hunk 6]
```diff
@@ -253,14 +253,14 @@ def add_result(self, result: ScenarioResult):
...
 
-    def process_results(self):
-        result = True
-        if len(self.results) == 0:
-            result = False
-        else:
-            for res in self.results:
-                if res.result is False:
-                    result = False
+    def process_results(self, result=True):
```
[the remark this answers is not in the document]
Comment (Nikhil-Singhal-06) [reply scenario_execution/scenario_execution/scenario_execution_base.py:256]: even if the output_dir parameter is passed?

[the remark this answers is not in the document]
Comment (fred-labs) [reply scenario_execution/scenario_execution/scenario_execution_base.py:256]: yes. you could print an info-message if output_dir was set in dry-run-mode.

File: scenario_execution_control/test/scenario_execution_control_test.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        sys.exit(1)
+    scenario_execute_control_test.execute_scenario(args.scenario)
+
+    start_time = time.time()
+    timeout = 60  # seconds
+    while rclpy.ok():
+        rclpy.spin_once(scenario_execute_control_test, timeout_sec=0.1)
+        if scenario_execute_control_test.future.done():
+            try:
+                scenario_execute_control_test.future.result()
```
Comment (fred-labs) [inline scenario_execution_control/test/scenario_execution_control_test.py:91]: this result should contain the scenario result, right? So it should be the main reasoning for the script return value. In addition you could make use of the scenario status as you've already done below.

File: scenario_execution_control/test/scenario_execution_control_test.py (hunk 5 above)
Comment (fred-labs) [reply scenario_execution_control/test/scenario_execution_control_test.py:45]: ok. then leave it in, but don't solely rely on the result (see comment below).

**Pull request closed without merging**